test/framework/src/main/java/org/elasticsearch/action/admin/indices/recovery/TransportRecoveryActionHelper.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.admin.indices.recovery;
+
+/**
+ * Helper methods for {@link TransportRecoveryAction}.
+ */
+public class TransportRecoveryActionHelper {
+
+    /**
+     * Helper method for tests to call {@link TransportRecoveryAction#setOnShardOperation}.
+     */
+    public static void setOnShardOperation(TransportRecoveryAction transportRecoveryAction, Runnable setOnShardOperation) {
```

Comment:
NIT: Maybe just inline this into the test class that actually uses it? Why have a separate method for this?

Reply:
I put it here so I could keep `TransportRecoveryAction#setOnShardOperation` package-private (without needing the test itself to be in the same package as the action)

Reply:
ah ok, makes sense :)